internal/intermediate/store/github.go

```diff
@@ -38,8 +38,12 @@ func (s *Store) GetGithubOAuthRedirectURL(ctx context.Context, req *intermediate
 		return nil, fmt.Errorf("enforce project login enabled: %w", err)
 	}
 
-	if qProject.GithubOauthClientID == nil {
-		return nil, apierror.NewFailedPreconditionError("github oauth client id not set", fmt.Errorf("github oauth client id not set"))
+	clientID := s.defaultGitHubOAuthClientID
+	redirectURI := s.defaultGitHubOAuthRedirectURI
+
+	if derefOrEmpty(qProject.GithubOauthClientID) != "" {
+		clientID = *qProject.GithubOauthClientID
+		redirectURI = req.RedirectUrl
```

Comment:
This logic only works because the redirect URI is always `<vaultDomain>/xxx-oauth-callback` right? Otherwise, this redirect URL is discarded and will be incorrect on the final redirect.

Reply:
That's right. We're locked into that regardless, because that URL pattern also exists in customer Google, Microsoft, and GitHub accounts on the IDP end.